### ISSUE TITLE
Metric functions

### ISSUE THREAD

**Issue by dakshitagrawal** [theEricMa/DiffSpeaker#6]
Hi, thank you for sharing the code.  I could not find the function definitions for `vocaset_upper_face_variance` and `vocaset_mouth_distance` that are imported at line 153 in `alm.models.modeltype.diffusion_bias`.

Could you please share the code for these functions? Thank you.

**Comment by Zavianx** [comment]
Hi, I use pretrained model for VOCASET datasets validation. This is my version. However, I found that the results were inconsistent with the paper results.
![image](https://github.com/theEricMa/DiffSpeaker/assets/36817799/ffb40fd8-b58a-4947-9c32-b33a453ac0c3)

```
exp = 'vocaset'
mouth_mask = list(range(94, 114)) + list(range(146, 178)) + list(range(183, 192))
pred = rs_set['vertice_pred'].squeeze().detach().cpu().numpy()
gt = rs_set['vertice'].squeeze().detach().cpu().numpy()
template =  batch['template'].squeeze().detach().cpu().numpy()

min_len = min(pred.shape[0], gt.shape[0])
pred = pred[:min_len, :]
gt = gt[:min_len, :]

gt_motion = gt
pred_motion = pred
lve = np.linalg.norm(pred_motion[:,mouth_mask] - gt_motion[:,mouth_mask], axis=1)

gt_std = np.std(np.square(gt),1)
pred_std = np.std(np.square(pred),1)
std_diff = np.abs(gt_std - pred_std)


metrics = {
    "FDD":torch.Tensor(std_diff).mean(),
    "Lip Vertex Error":torch.Tensor(lve).mean(),
    "Length": torch.tensor(min_len).float(),
}
```

**Comment by theEricMa** [comment]
Dear all, I have uploaded the evaluation code. The metrics are consistent with those reported in the paper on my computer. I found the model weight of `diffspeaker_hubert_vocaset.ckpt` is the wrong version, and I updated it with the right one. Please have a try then :)

**Comment by Zavianx** [comment]
Thanks, I have reproduced your results successfully. How many training epoch did this result take?

**Comment by theEricMa** [comment]
The config files in `configs/diffusion` specify the number of training iterations. The BIWI dataset experiments utilized 8 GPUs, leading to fewer epochs compared to those needed for the vocaset experiments. Let me close this issue since the problem is solved.
